concent_api/core/views.py

```diff
@@ -575,6 +661,14 @@ def handle_send_force_subtask_results(request, client_message: message.concents.
             reason      = message.concents.ServiceRefused.REASON.DuplicateRequest,
         )
 
+    if Subtask.objects.filter(
+        subtask_id = client_message.ack_report_computed_task.task_to_compute.compute_task_def['subtask_id'],
+        state      = Subtask.SubtaskState.FORCING_ACCEPTANCE.name,  # pylint: disable=no-member
```

Comment:
Filter only by `subtask_id` here. Then check `state`. Otherwise the `else` covers cases that are not really duplicate requests.

Reply:
Same as above, I will copy for clarification

I think that is exactly the case we want to cover here. 
There might be a `Subtask` with given `subtask_id` but in different state - then we proceed and code fails on state transition if it is forbidden.
There might be a `Subtask`s with given `FORCING_RESULT_TRANSFER` state but with different `subtask_id` - then we proceed.
When there is a `Subtask` in `FORCING_RESULT_TRANSFER` state with given `subtask_id`, it is duplicated request.
What did I get wrong here ?

Reply:
Same as above. You're right. The code is fine.